lib/spack/spack/directory_layout.py

```diff
@@ -264,13 +264,18 @@ def create_install_directory(self, spec):
         from spack.package_prefs import get_package_group
         group = get_package_group(spec)
         perms = get_package_dir_permissions(spec)
-        mkdirp(spec.prefix, mode=perms)
+        perms_intermediate = 0o777
```

Comment:
Can you add a comment here to explain what this is doing? Maybe like a 2-3 line summary of our discussion?

Reply:
Yeah, sure! it deserves that!

Reply:
Yeah, sure! it deserves that!